File: server_curve_fever/Point.cpp
#include "Point.h"

Point::Point(double x, double y) {
    this->x = x;
    this->y = y;
    this->gap = false;
}

Point::Point(double x, double y, bool gap) {
    this->x = x;
    this->y = y;
    this->gap = gap;
}

void Point::setGap(bool gap) {
    this->gap = gap;
}

bool Point::isGap() {
    return gap;
}

double Point::getX() {
    return x;
}

double Point::getY() {
    return y;
}

File: client_curve_fever/src/pl/poznan/put/game/Point.java
package pl.poznan.put.game;

public class Point {
    private int number;
    private double x;
    private double y;
    private boolean gap;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Point(double x, double y, boolean gap) {
        this.x = x;
        this.y = y;
        this.gap = gap;
    }

    public boolean isGap() {
        return gap;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }
}